grp/classic.gi

```diff
@@ -1917,6 +1917,122 @@ InstallMethod( Omega,
     { filt, e, d, R } -> OmegaCons( filt, e, d, Size( R ) ) );
 
 
+#############################################################################
+##
+#F  WallForm( <form>, <m>, <fld> ) . . . . . . . . . . . compute the wall of <m>
+##
+# Input: <m> is an element of the orthogonal group which is given by the form <form>,
+#        <fld> is the finite field of the orthogonal group (in which <m> is defined)
+# Output: The Wall Form of <m> correpsponding to <form>
+# Computes the Wall Form of a matrix. The definition can be found in
+# [Taylor, page 163].
+BindGlobal( "WallForm", function( form, m, fld )
+    local id,  w,  b,  p,  i,  x,  j, d;
+
+    id := One( m );
+
+    # compute a base for Image(id-m) which is a subset of the rows of (id - m)
+    # We also store the index of the rows (corresponding to w) in p
+    w := id - m;
+    b := [];
+    p := [];
+    for i in [ 1 .. Length(w) ]  do
+        if Length(b) = 0  then
+            if w[i] <> 0*w[i]  then
+                Add( b, w[i] );
+                Add( p, i );
+            fi;
+        elif RankMat(b) <> RankMat(Concatenation(b,[w[i]]))  then
+            Add( b, w[i] );
+            Add( p, i );
+        fi;
+    od;
+
+    # compute the form
+    d := Length(b);
+    x := NullMat(d,d,fld);
+    for i  in [ 1 .. d ]  do
+        for j  in [ 1 .. d ]  do
+            x[i,j] := form[p[i]] * b[j];
+        od;
+    od;
+
+    # and return
+    return rec( base := b, pos := p, form := x );
+
+end );
+
+
+#############################################################################
+##
+#F  IsSquareWithoutZeroFFE( fld, e) . . . . . . . . . . . Tests whether <e> (not zero) is a
```

Comment:
How about `IsNonZeroSquareFFE` ?

But why do you need this anyway? Your determinants are always non-zero, aren't they?